common/src/main/java/io/novafoundation/nova/common/validation/FieldValidator.kt

```diff
@@ -0,0 +1,17 @@
+package io.novafoundation.nova.common.validation
+
+import kotlinx.coroutines.flow.Flow
+
+interface FieldValidator {
+    fun observe(inputStream: Flow<String>): Flow<FieldValidationResult>
+
+    companion object
+}
```

Comment:
What do you think about making a small subclass named something like `MapFieldValidator`?

``` kotlin

class MapFieldValidator : FieldValidator {
  abstract fun validate(input: String): FieldValidationResult

 override fun observe(inputStream: Flow<String>) = inputStream.map(::validate)
}
```

This will reduce boilerplate for simpler validations like slippage one